### ISSUE TITLE
transition to "primary/replica" instead of "master/slave"

### ISSUE THREAD

**Issue by clonm** [cloyne/docker-powerdns#1]
https://en.wikipedia.org/wiki/Master/slave_(technology)#Terminology_concerns
As of December 2017, this alternative terminology is an accepted open standard for BIND. But, updating will require some care to make sure we don't break our nameservers.

I changed the naming things that I'm sure are trivial:
https://github.com/cloyne/docker-powerdns-secondary/commit/265924eb7c4b269715d303f1a571424384814841
https://github.com/cloyne/servers/commit/739245eb660d79d833aabeaae368c24127c58d3b
https://cloud.docker.com/u/cloyne/repository/docker/cloyne/powerdns-secondary

But, I'm not sure if it's safe to change the actual config files. In addition, the version of PowerDNS we use is no longer supported. Since [the most recent version still doesn't have an alternative for "slave" in the options](https://doc.powerdns.com/authoritative/settings.html) and logging seems to be broken anyway, it might make sense to look into an alternative instead of upgrading to 4.x.

**Comment by mitar** [comment]
I do not think we use bind?

**Comment by clonm** [comment]
@mitar I thought you [said we did](https://github.com/cloyne/network/issues/88#issuecomment-284570720)? 
Also:
[pdns.cloyne](https://github.com/cloyne/docker-powerdns/blob/master/etc/powerdns/pdns.d/pdns.cloyne)
```
# Check interval.
bind-check-interval=3600
```
[bindbackend.conf](https://github.com/cloyne/docker-powerdns-primary/blob/master/etc/powerdns/bindbackend.conf)
[etc/powerdns/bind/](https://github.com/cloyne/docker-powerdns-primary/tree/master/etc/powerdns/bind)

I uncommented `# launch=bind` and add `bind-config=/etc/powerdns/bindbackend.conf` in `/etc/powerdns/pdns.conf` to get the service to restart on server2 (without that, it was saying `Unable to launch, no backends configured for querying`). It's still logging in the wrong place but it had (on its own) stopped logging at all and wasn't responsive to `pdns_control status` from within the container. If we aren't using BIND though, this was probably not the right fix...

**Comment by mitar** [comment]
No, we are using bind-backend which allows PowerDNS to use bind configuration/zone files. So bind zone files are so common that our DNS server supports using them. Because they are common, I am familiar with them so it was an easy solution for configuration to store in files.

**Comment by mitar** [comment]
For PowerDNS it seems they are still using master/slave terminology: https://doc.powerdns.com/authoritative/modes-of-operation.html

You can always try to submit a patch to them. Or, patch just our Docker image. Depending how much you care about this.

**Comment by clonm** [comment]
Thanks for explaining the difference. The version of PowerDNS we use is no longer supported anyway - are there alternatives to PowerDNS we might be able to use instead instead of upgrading? Or is that even more complicated than submitting an upstream patch (which I probably won't do since I understand it's still controversial within the community and I don't have that kind of clout) (yet).

**Comment by mitar** [comment]
So do you know if it is controversial in that particular community? Maybe just nobody brought it up.

I like PowerDNS otherwise. It is a pretty good tool. You might want to update to bind though. You already have all configuration. :-)
